setuptools/tests/test_easy_install.py

```diff
@@ -1332,6 +1332,22 @@ def test_from_simple_string_uses_shlex(self):
         assert len(cmd) == 2
         assert '"' not in cmd.as_header()
 
+    def test_from_param_raises_expected_error(self) -> None:
+        """
+        from_param should raise its own TypeError when the argument's type is unsupported
+        """
+        try:
+            ei.CommandSpec.from_param(object())  # type: ignore[arg-type] # We want a type error here
+        except Exception as error:
+            assert type(error) is TypeError, error
+            assert (
+                str(error) == "Argument has an unsupported type <class 'object'>"
+            ), error
+        else:
+            raise AssertionError(
+                "from_param did not raise any error for argument of unsupported type"
+            )
+
```

Comment:
For coverage purposes: https://github.com/pypa/setuptools/actions/runs/10305775968/job/28527394123?pr=4505#step:5:74

Reply:
https://github.com/pypa/setuptools/actions/runs/10307637126/job/28533318670?pr=4548#step:5:74
> setuptools/tests/test_easy_install.py (85.7%): Missing lines 1347

Darn, coverage check includes the test themselves. Is there a better way to do a "expect to fail with specific error" test ?

Reply:
Yeah, it can be a bit overwhelming.

> Is there a better way to do a "expect to fail with specific error" test ?

Have you tried [`pytest.raises`](https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions)?